src/aatoolbox/datasources/ecmwf_api/ecmwf_api_download_seasonal_monthly.py

```diff
@@ -61,45 +66,74 @@ def download_date(
     ----------
     iso3 : str
         iso3 code of country of interest
-    date_forec : Union[str, date]
-        date the forecast is released. Only the year and month
-        are of importance
     ecmwf_dir : Union[str, Path]
         path to the ecmwf dir to which the data should be written
     iso3_gdf : gpd.GeoDataFrame
         GeoDataFrame which contains geometries describing the area
+    date_list : List[Union[str, date]]
+        list of dates the forecasts should be downloaded for.
+        If None, all forecasts are downloaded
     area : Area, default = None
         Area object containing the boundary coordinates of the area that
         should be downloaded. If None, retrieved from iso3_gdf
+    grid: float, default = 0.4
+        Grid resolution in degrees
 
     Examples
     --------
     >>> import geopandas as gpd
     >>> df_admin_boundaries = gpd.read_file(gpd.datasets.get_path('nybb'))
-    >>> download_date(iso3="nybb",date_forec=date(1999,1,1),
-    ... ecmwf_dir=ecmwf_dir,iso3_gdf=df_admin_boundaries.to_crs("epsg:4326"))
+    >>> download(iso3="nybb",ecmwf_dir=ecmwf_dir,
+    ... iso3_gdf=df_admin_boundaries.to_crs("epsg:4326"))
     """
-    if not isinstance(date_forec, date):
-        date_forec = date.fromisoformat(date_forec)
-
-    if not isinstance(ecmwf_dir, Path):
-        ecmwf_dir = Path(ecmwf_dir)
-
     # retrieve coord boundaries for which to download data
     if not area:
         area = AreaFromShape(iso3_gdf)
         # prefer to round the coordinates to integers as this
         # will lead to more correspondence to the grid that ecmwf
         # publishes its data on
         area.round_area_coords()
-
-    # TODO: it would probably be safer to also include the boundary coords
-    # in the filename.. Just that the filename then gets massive
-    output_filename = (
-        f"{iso3}_{SEAS_DIR}_{PRATE_DIR}_{date_forec.strftime('%Y-%m')}.nc"
-    )
-    output_path = ecmwf_dir / SEAS_DIR / PRATE_DIR / output_filename
-    output_path.parent.mkdir(exist_ok=True, parents=True)
+    if date_list is None:
+        # TODO: the end date should be updated dynamically
+        # or catch the APIException for dates that don't exist
+        date_list = [
+            d.strftime("%Y-%m-%d")
+            for d in pd.date_range(
+                start="1992-01-01", end="2021-04-01", freq="MS"
+            )
+        ]
+    for date_forec in date_list:
+        if not isinstance(date_forec, date):
+            date_forec = date.fromisoformat(date_forec)
+        # TODO: it would probably be safer to also include the boundary coords
```

Comment:
hm actually that's not a bad idea. If the filenames are all "under the hood" then the length doesn't matter too much. It's just tricky if the coords are long floats.

Reply:
Maybe we could round them to two decimals?

Reply:
And then also round the coords to two decimals before making the API request?